### PR TITLE
Added Support AWS Secrets Manager Vault from Camel

### PR DESCRIPTION
**Pull request by oscerd**
This is a first addon for #3031

### CODE REVIEW

**Review by orpiske — APPROVED** [review]

File: addons/vault/aws_secrets_manager.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		for _, endpoint := range meta.FromURIs {
+			log.Infof("Processing component %s", endpoint)
+		}
+	}
+
+	if r.UseDefaultCredentialsProvider == "" {
+		r.UseDefaultCredentialsProvider = "false"
+	}
+
+	return r.Enabled != nil && *r.Enabled, nil
```
Comment (astefanutti) [inline addons/vault/aws_secrets_manager.go:97]: This is always truthy at that point, so could be `return true, nil` for clarity.

File: addons/vault/aws_secrets_manager.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+		meta, err := metadata.ExtractAll(environment.CamelCatalog, sources)
+		if err != nil {
+			return false, err
+		}
+
+		for _, endpoint := range meta.FromURIs {
+			log.Infof("Processing component %s", endpoint)
+		}
+	}
```
Comment (astefanutti) [inline addons/vault/aws_secrets_manager.go:91]: Is that section actually needed? It seems it only logs the endpoints?

Comment (oscerd) [reply addons/vault/aws_secrets_manager.go]: Nope, I still need to find my dimension with golang and with the camel-k project, so my code will be a mess :-D

File: addons/vault/aws_secrets_manager.go (hunk 1 above)
Comment (oscerd) [reply addons/vault/aws_secrets_manager.go]: It should be fine now.

**Comment by oscerd** [comment]
If we all agree on this, I'd like to backport the addon to 1.10.x, since the feature in the components for doing is available from < 3.18.x camel releases. Only the hashicorp component will come with Camel 3.19.0, so we could have the vault support for gcp, aws and azure even in 1.10.x.

**Review by orpiske — APPROVED** [review]

**Comment by orpiske** [comment]
> If we all agree on this, I'd like to backport the addon to 1.10.x, since the feature in the components for doing is available from < 3.18.x camel releases. Only the hashicorp component will come with Camel 3.19.0, so we could have the vault support for gcp, aws and azure even in 1.10.x.

+1 from me.

File: addons/vault/aws_secrets_manager.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+		util.StringSliceUniqueAdd(&environment.Integration.Status.Capabilities, v1.CapabilityAwsSecretsManager)
+		// Add the Camel Quarkus AWS Secrets Manager
+		util.StringSliceUniqueAdd(&environment.Integration.Status.Dependencies, "mvn:org.apache.camel.quarkus:camel-quarkus-aws-secrets-manager")
+	}
+
+	if environment.IntegrationInRunningPhases() {
+		environment.ApplicationProperties["camel.vault.aws.accessKey"] = r.AccessKey
+		environment.ApplicationProperties["camel.vault.aws.secretKey"] = r.SecretKey
+		environment.ApplicationProperties["camel.vault.aws.region"] = r.Region
+		environment.ApplicationProperties["camel.vault.aws.defaultCredentialsProvider"] = r.UseDefaultCredentialsProvider
```
Comment (astefanutti) [inline addons/vault/aws_secrets_manager.go:91]: From https://camel.apache.org/components/next/aws-secrets-manager-component.html#_component_options, should that be `usedefaultCredentialsProvider`

File: addons/vault/aws_secrets_manager.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	Auto *bool `property:"auto" json:"auto,omitempty"`
+	// The AWS Access Key to use
+	AccessKey string `property:"access-key,omitempty"`
+	// The AWS Secret Key to use
+	SecretKey string `property:"secret-key,omitempty"`
+	// The AWS Region to use
+	Region string `property:"region,omitempty"`
+	// The adapter-specific policy to use when filling the cache (use: minimizing / maximizing). Check
+	// the component documentation if unsure
+	UseDefaultCredentialsProvider string `property:"use-default-credentials-provider,omitempty"`
```
Comment (astefanutti) [inline addons/vault/aws_secrets_manager.go:50]: Could that be a `*bool`?

File: addons/vault/aws_secrets_manager.go (hunk 2 above)
Comment (astefanutti) [reply addons/vault/aws_secrets_manager.go]: 😄, no pb!

File: addons/vault/aws_secrets_manager.go (hunk 3 above)
Comment (oscerd) [reply addons/vault/aws_secrets_manager.go:91]: It's defined in the vault configuration: https://github.com/apache/camel/blob/main/core/camel-main/src/generated/resources/META-INF/camel-main-configuration-metadata.json#L212

It's not consistent, we should fix this for sure. Thanks for spotting.

File: addons/vault/aws_secrets_manager.go (hunk 4 above)
Comment (oscerd) [reply addons/vault/aws_secrets_manager.go:50]: yes, but the application.properties would expect a string anyway, but we could make it a string while assigning. Let me know.

Comment (astefanutti) [reply addons/vault/aws_secrets_manager.go:50]: You could do something like:

```go
environment.ApplicationProperties["camel.vault.aws.defaultCredentialsProvider"] = strconv.FormatBool(r.UseDefaultCredentialsProvider)
```

I think that's better being more constrained if possible, as it's part of the end-user API.

Comment (oscerd) [reply addons/vault/aws_secrets_manager.go:50]: Then it should be a bool and not a *bool

File: addons/vault/aws_secrets_manager.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+	Trait `property:",squash"`
+}
+
+func NewAwsSecretsManagerTrait() trait.Trait {
+	return &awsSecretsManagerTrait{
+		BaseTrait: trait.NewBaseTrait("aws-secrets-manager", trait.TraitOrderBeforeControllerCreation),
+	}
+}
+
+func (r *awsSecretsManagerTrait) Configure(environment *trait.Environment) (bool, error) {
```
Comment (astefanutti) [inline addons/vault/aws_secrets_manager.go:64]: nit: `func (t *awsSecretsManagerTrait)`

File: addons/vault/aws_secrets_manager.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+	}
+
+	if r.UseDefaultCredentialsProvider == "" {
+		r.UseDefaultCredentialsProvider = "false"
+	}
+
+	return true, nil
+}
+
+func (r *awsSecretsManagerTrait) Apply(environment *trait.Environment) error {
```
Comment (astefanutti) [inline addons/vault/aws_secrets_manager.go:80]: nit: `func (t *awsSecretsManagerTrait)`

File: addons/vault/aws_secrets_manager.go (hunk 4 above)
Comment (astefanutti) [reply addons/vault/aws_secrets_manager.go:50]: The `Configure` method could also make sure to set a default value if the pointer is nil, as it currently set it to `"false"`.

File: addons/vault/aws_secrets_manager.go (hunk 5 above)
Comment (oscerd) [reply addons/vault/aws_secrets_manager.go]: Done

File: addons/vault/aws_secrets_manager.go (hunk 6 above)
Comment (oscerd) [reply addons/vault/aws_secrets_manager.go]: Done

**Pull request merged**